### PR TITLE
Cylance - Delete device from cylance, rework utility function

### PR DESCRIPTION
**Pull request by ncramer-r7**
### Description

Describe the proposed changes:

  - Add the delete asset action in Cylance
  - Move functions into utility folder (affects quarantine action)
  - Add unit test templates

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

docker run --rm -i rapid7/cylance_protect:1.5.0 run < tests/quarantine.json
```
{
  "lockdown_details": {
    "data": {
      "connection_status": "connected",
      "hostname": "VAGRANT-PC",
      "id": "E152489F18C842F2B0B1907E24FC0A94",
      "lockdown_expiration": "2020-09-18T20:00:43Z",
      "lockdown_initiated": "2020-09-15T20:00:43Z",
      "optics_device_version": "2.5.1100.1139",
      "tenant_id": "b9b35c46538b4544b2d16200ecdb64dd"
    },
    "status": "COMPLETE"
  }
}
```

docker run --rm -i rapid7/cylance_protect:1.5.0 run < tests/delete_asset.json

```
{
  "deleted": [
    "10.0.2.15"
  ],
  "not_deleted": [],
  "success": true
}
```

### Validation
```
ncramer@BOS-MBP-8176:~/go/src/github.com/rapid7/insightconnect-plugins/cylance_protect|delete-asset-cylance ⇒  icon-validate .
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Executing validator RuntimeValidator
[*] Plugin successfully validated!
```

### CODE REVIEW

File: cylance_protect/help.md [hunk 1]
```diff
@@ -48,6 +49,54 @@ Example input:
...
+|----|----|--------|-----------|
+|deleted|[]string|False|List of assets that were deleted|
+|not_deleted|[]string|False|List of assets that were not deleted, either because of whitelist or because they were not found|
+|success|boolean|True|Return true if at least one device was deleted|
+
+Example output:
+
+```
+{
+    "deleted": ["10.0.2.15"],
```
Comment (jschipp-r7) [inline cylance_protect/help.md:91]: These should be 2 space indention

File: cylance_protect/help.md [hunk 2]
```diff
@@ -64,14 +113,9 @@ Example input:
 
 ```
 {
-  "add_zones": [
```
Comment (jschipp-r7) [inline cylance_protect/help.md:67]: Were these supposed to be removed?

File: cylance_protect/help.md [hunk 3]
```diff
@@ -434,6 +478,7 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 1.5.0 - New action Delete Asset, rework utility function
```
Comment (jschipp-r7) [inline cylance_protect/help.md:481]:
```suggestion
* 1.5.0 - New action Delete Asset | Rework utility function
```

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]

File: cylance_protect/help.md (hunk 2 above)
Comment (ncramer-r7) [reply cylance_protect/help.md:67]: Yes! It seems like the last update to this plugin had an input example that did match the spec of the action...this was generated from regenerate

File: cylance_protect/help.md [hunk 4]
```diff
@@ -48,6 +49,56 @@ Example input:
 
 ### Actions
 
+#### Delete Asset
+
+This action is used to delete assets/devices from the Console.
```
Comment (jschipp-r7) [inline cylance_protect/help.md:54]:
```suggestion
This action is used to delete assets/devices from the Cylance console.
```

**Review by jschipp-r7 — APPROVED** [review]

**Pull request merged**